MicrosoftActiveDirectory/tests/test_search.py

```diff
@@ -0,0 +1,30 @@
+from search_in_ad_modules.base import MicrosoftADModule, MicrosoftADAction
+from search_in_ad_modules.search import SearchAction
+from unittest.mock import patch
+import pytest
+
+
+def configured_action(action: MicrosoftADAction):
+    module = MicrosoftADModule()
+    a = action(module)
+
+    a.module.configuration = {
+        "servername": "test_servername",
+        "admin_username": "test_admin_username",
+        "admin_password": "test_admin_password",
+    }
+
+    return a
+
+
+@pytest.fixture
+def one_user_dn():
+    return [["CN=integration_test,CN=Users,DC=lab,DC=test,DC=com", 512]]
+
+def test_search():
```

Comment:
I'm not sure that this test will pass. I think you need to mock the search method.